src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -0,0 +1,84 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.Models;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export
+{
+    public class ExportJobTask
+    {
+        private readonly ExportJobRecord _exportJobRecord;
+        private readonly IFhirOperationsDataStore _fhirOperationsDataStore;
+        private readonly ILogger _logger;
+
+        private WeakETag _weakETag;
+
+        public ExportJobTask(
+            ExportJobRecord exportJobRecord,
+            WeakETag weakETag,
+            IFhirOperationsDataStore fhirOperationsDataStore,
+            ILogger<ExportJobTask> logger)
+        {
+            EnsureArg.IsNotNull(exportJobRecord, nameof(exportJobRecord));
+            EnsureArg.IsNotNull(weakETag, nameof(weakETag));
+            EnsureArg.IsNotNull(fhirOperationsDataStore, nameof(fhirOperationsDataStore));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _exportJobRecord = exportJobRecord;
+            _fhirOperationsDataStore = fhirOperationsDataStore;
+            _logger = logger;
+
+            _weakETag = weakETag;
+        }
+
+        public async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Try to acquire the job.
+                try
+                {
+                    _logger.LogTrace("Acquiring the job.");
+
+                    await UpdateJobStatus(OperationStatus.Running, cancellationToken);
+                }
+                catch (JobConflictException)
+                {
+                    // The job is taken by another process.
+                    _logger.LogWarning("Failed to acquire the job. The job was acquired by another process.");
+                    return;
+                }
+
+                // We have acquired the job, process the export.
```

Comment:
Dumb question: this is TODO, right?

Reply:
Yes. I forgot to link back to the work item, which would have explained the scope of this PR. The work item is here: AB#68716.